src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsAdornmentManager.cs

```diff
@@ -43,6 +43,11 @@ internal class InlineDiagnosticsAdornmentManager : AbstractAdornmentManager<Inli
             _endLineTagAggregator.BatchedTagsChanged += EndLineTagAggregator_BatchedTagsChanged;
         }
 
+        /// <summary>
+        /// Getting all tags changed events and removing all inline diagnostics to be redrawn
+        /// based on if they intersect with any IEndOfLineAdornmentTags after the layout change
+        /// has completed.
+        /// </summary>
```

Comment:
In the loop below, aggregating all the changedSpans into a collection of everything that changed and then calling UpdateSpans... on that might be more efficient (I'm assuming changedSpans is something similar to the editor's NormalizedSpanCollection).

Reply:
now flatten the list of spans